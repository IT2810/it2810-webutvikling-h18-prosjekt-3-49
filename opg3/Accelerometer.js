import React from 'react';
import {Accelerometer} from "expo";
import {Text, View} from "react-native";
import roundTo from 'round-to';


export default class Acc extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            accelerometerData: {},
        };
    }

    componentDidMount() {
        this._toggle();
    }

    componentWillUnmount() {
        this._unsubscribe();
    }

    _toggle = () => {
        if (this._subscription) {
            this._unsubscribe();
        } else {
            this._subscribe();
        }
    };

    _subscribe = () => {
        this._subscription = Accelerometer.addListener(accelerometerData => {
            this.setState({ accelerometerData });
        });
    };

    _unsubscribe = () => {
        this._subscription && this._subscription.remove();
        this._subscription = null;
    };

    render() {
        let {x, y, z} = this.state.accelerometerData;
        if (x !== undefined && y !== undefined && z !== undefined) {
            x = roundTo(x, 1);
            y = roundTo(y, 1);
            z = roundTo(z, 1);
        }
        if (Math.abs(x) >= Math.abs(y) && Math.abs(x) >= Math.abs(z)) {
            return (
                <View>
                    <Text style={{fontWeight: 'bold'}}>Accelerometer demo:</Text>
                    <Text style={{backgroundColor: '#66DF59'}}>It seems that your phone is in landscape-mode!</Text>
                </View>
            )
        }

        if (Math.abs(y) >= Math.abs(x) && Math.abs(y) >= Math.abs(z)) {
            return (
                <View>
                    <Text style={{fontWeight: 'bold'}}>Accelerometer demo:</Text>
                    <Text style={{backgroundColor: '#6D6FEF'}}>It seems that your phone is in portrait-mode!</Text>
                </View>
            )
        }

        if (Math.abs(z) >= Math.abs(x) && Math.abs(z) >= Math.abs(y)) {
            return (
                <View>
                    <Text style={{fontWeight: 'bold'}}>Accelerometer demo:</Text>
                    <Text style={{backgroundColor: '#5CAACC'}}>It seems that your phone is lying down!</Text>
                </View>
            )
        }

        return (
            <View>
                <Text style={{fontWeight: 'bold'}}>Accelerometer demo:</Text>
                <Text>Values are undefined</Text>
            </View>
        )
    }
}
