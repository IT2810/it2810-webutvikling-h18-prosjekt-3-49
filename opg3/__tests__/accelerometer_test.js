import React from 'react';
import Accelerometer from "../components/Accelerometer";

import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer.create(<Accelerometer />).toJSON();
    expect(tree).toMatchSnapshot();
});