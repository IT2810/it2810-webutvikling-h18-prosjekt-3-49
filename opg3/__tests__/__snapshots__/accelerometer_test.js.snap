// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    Accelerometer demo:
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Values are undefined
  </Text>
</View>
`;
